src/subscription_manager/cli_command/register.py

```diff
@@ -462,9 +466,16 @@ def _get_owner_cb(self, owners):
             )
         )
 
-        # Read the owner key from stdin
+        # Read the owner key from stdin or raise a system error if in a non-interactive session.
+        if not self._is_atty():
+            raise system_exit(os.EX_USAGE, "Error: --org parameter not provided!")
+
         owner_key = None
         while not owner_key:
             owner_key = input(_("Organization: "))
             readline.clear_history()
         return owner_key
+
+    @staticmethod
+    def _is_atty():
+        return sys.stdin and sys.stdin.isatty()
```

Comment:
this is a good idea, however the same logic is duplicated in user_pass.py; hence, i'd move it to a more generic place such as src/subscription_manager/utils.py, with a more fitting name eg "is_interactive()"